packages/validator/src/services/blockDuties.ts

```diff
@@ -182,4 +183,22 @@ export class BlockDutiesService {
       }
     }
   }
+
+  private async isIncludedInCanonicalChain(slot: Slot): Promise<boolean> {
+    //#TODO: figure out how to get the latest proposed block and compare it to the canonical block
```

Comment:
As noted above implementing this function is not easy and will rely on probabilistic assumptions. The beacon chain has complete info about the chain it should be done there

Reply:
Good point